src/backend/distributed/connection/connection_management.c

```diff
@@ -694,8 +715,15 @@ StartConnectionEstablishment(ConnectionHashKey *key)
 	entry = hash_search(ConnParamsHash, key, HASH_ENTER, &found);
 	if (!found || !entry->isValid)
 	{
-		/* if they're not found, compute them from GUC, runtime, etc. */
-		GetConnParams(key, &entry->keywords, &entry->values, ConnectionContext);
+		/* avoid leaking memory in the keys and values arrays */
+		if (found && !entry->isValid)
+		{
+			FreeConnParamsHashEntryFields(entry);
```

Comment:
This probably shouldn't hold up the PR, but I kind of wonder what happens if there is an out of memory error thrown in `GetConnParams`. We'll re-enter this code, and then most likely segfault by trying to free twice.

Reply:
>  but I kind of wonder what happens if there is an out of memory error thrown in GetConnParams.

Just to clarify: If we get out of memory error, wouldn't the execution end (or broadly transaction fail)? How could we re-enter this code?

Reply:
on the next transaction?

Reply:
I attempted a mitigation for this but am worried about edge cases. Giving up and I think we need to resurrect my `pg_fiu` extension to start injecting OOM errors into our codebase to shake this stuff out.